scripts/Muon/GUI/Common/plotting_widget/plotting_widget_presenter.py

```diff
@@ -431,6 +432,113 @@ def plot_all_selected_workspaces(self, autoscale=False):
+        self.update_options_view_from_model()
+        self._view.force_redraw()
+
+    def handle_plot_single_sequential_fit(self, selected_workspaces):
+        if not selected_workspaces:  # check selected_workspaces is empty
+            return
+
+        errors = self._view.plot_options.get_errors()
+        fitted_workspace_list = []
+        workspace_list = []
+        # get corresponding fit for the workspace_list, if it exists
+        for _, workspaces in selected_workspaces.items():
+            workspace_list += workspaces
+            fit = self.context.fitting_context.find_fit_for_input_workspace_list(workspaces)
+            if fit is not None:
+                fitted_workspace_list += fit.output_workspace_names
+            else:
+                fitted_workspace_list += []
+
+        for ws in self._model.plotted_workspaces:
+            if ws not in workspace_list:
+                self._model.remove_workspace_from_plot(ws, self._view.get_axes())
+
+        for ws in self._model.plotted_fit_workspaces:
+            if ws not in fitted_workspace_list:
+                self._model.remove_workspace_from_plot(ws, self._view.get_axes())
+
+        for workspace in workspace_list:
+            if workspace not in self._model.plotted_workspaces:
+                ax = self.get_workspace_plot_axis(workspace)
+                label = self.get_workspace_legend_label(workspace)
+                self._model.add_workspace_to_plot(ax, workspace, [0], errors=errors,
+                                                  plot_kwargs={'distribution': True, 'autoscale_on_update': False,
+                                                               'label': label})
+                self._model.add_workspace_to_plotted_workspaces(workspace)
+
+        for workspace_name in fitted_workspace_list:
+            if workspace_name in self._model.plotted_fit_workspaces:
+                continue
+
+            label = self.get_workspace_legend_label(workspace_name)
+            ax = self.get_workspace_plot_axis(workspace_name)
+            # Get fit function name, as we are only ever storing 1 fit in this context
+            # we can take the first entry
+            fit_function = self.context.fitting_context.fit_function_names()[0]
+            self._model.add_workspace_to_plotted_fit_workspaces(workspace_name)
+
+            # handle tf asymmetry fits
+            first_fit_index = 1
+            if TF_ASYMMETRY_PREFIX in workspace_name:
+                first_fit_index = 3
+
+            self._model.add_workspace_to_plot(ax, workspace_name, [first_fit_index],
+                                              errors=False,
+                                              plot_kwargs={'distribution': True, 'autoscale_on_update': False,
+                                                           'label': label + fit_function + ': Fit'})
+            self._model.add_workspace_to_plot(ax, workspace_name, [2],
```

Comment:
there seems to be some repeated code in this file. Can we pull it out into functions? e.g. ```
def plot_TF_Asymm:
    if .....
```